src/core/fusebox.ts

```diff
@@ -1,26 +1,27 @@
+import { EnvironmentType } from '../config/EnvironmentType';
 import { IPublicConfig } from '../config/IConfig';
 import { IRunProps } from '../config/IRunProps';
 import { bundleDev } from '../development/bundleDev';
 import { bundleProd } from '../production/bundleProd';
-
-import { EnvironmentType } from '../config/EnvironmentType';
 import { createContext, ICreateContextProps } from './context';
-import { finalizeFusebox } from './helpers/finalizeFusebox';
 import { preflightFusebox } from './helpers/preflightFusebox';
 
 export function fusebox(publicConfig: IPublicConfig) {
-  function execute(props: ICreateContextProps) {
+  async function execute(props: ICreateContextProps) {
     const ctx = createContext(props);
+
+    ctx.isWorking = true;
+    ctx.ict.sync('init', { ctx: ctx });
     preflightFusebox(ctx);
     switch (props.envType) {
```

Comment:
also add a comment here with `//todo try {} catch`

Reply:
done, i"ve wrapped it into a try catch block